### PR TITLE
Adding code to enable nbd for all networks on the host

### PR DESCRIPTION
**Pull request by thomasmck**
Signed-off-by: Thomas Mckelvey <thomas.mckelvey@citrix.com>

### CODE REVIEW

File: cbt_enable_and_snapshot.py [hunk 1]
```diff
@@ -20,6 +20,14 @@
 import argparse
 
 
+def enable_NBD(session):
+    # Not sure if we want to make nbd/nbd_insecure an option here
```
Comment (thomasmck) [inline cbt_enable_and_snapshot.py]: This is the only question I have for this commit

Comment (robertbreker) [reply cbt_enable_and_snapshot.py]: I don't think we need this, as this example is showing the better (secure) variant. Would suggest to remove this comment.

File: cbt_enable_and_snapshot.py [hunk 2]
```diff
@@ -46,6 +54,7 @@ def main():
                                 "CBT example")
 
     try:
+        enable_NBD(session)
```
Comment (robertbreker) [inline cbt_enable_and_snapshot.py]: Minor:
Should we rename this to enable_nbd_on_all_networks()?
Also function names should be lowercase according to pep8.

File: cbt_enable_and_snapshot.py [hunk 3]
```diff
@@ -20,6 +20,14 @@
 import argparse
 
 
+def enable_NBD(session):
```
Comment (robertbreker) [inline cbt_enable_and_snapshot.py]: Can you add a description for what you're doing in the header?

File: cbt_enable_and_snapshot.py [hunk 4]
```diff
@@ -6,6 +6,10 @@
 backups and will create the base snapshot. After the VDI has been exported
 the snapshot data is destroyed to save space on the host.
 
+The enable_nbd_on_all_networks function will add the nbd option to the 
+purpose field of all networks on the host you are connecting to thereby
+enablinh nbd connections on all networks.
```
Comment (robertbreker) [inline cbt_enable_and_snapshot.py]: Minor: enablinh

File: cbt_enable_and_snapshot.py [hunk 5]
```diff
@@ -20,6 +24,13 @@
 import argparse
 
 
+def enable_nbd_on_all_networks(session):
+    connection_type = "nbd"
+    network_list = (session.xenapi.network.get_all())[0].split(",")
```
Comment (robertbreker) [inline cbt_enable_and_snapshot.py]: This is wrong. Sorry for not spotting this earlier.

session.xenapi.network.get_all() will return references to all networks.
[0] will return the first reference
split() will convert this back into an array.

II think you just want session.xenapi.network.get_all() to get all networks?!

Comment (thomasmck) [reply cbt_enable_and_snapshot.py]: Yes you are right. Copied this code from my XenRT test and seeing as it didn't fail I had just assumed it was working. Fixed now.

**Pull request merged**